### PR TITLE
Disable the `requestLegacyExternalStorage` attribute when there are no storage permissions

### LINKED ISSUE #47910: Use of All files access (MANAGE_EXTERNAL_STORAGE) permission
**Godot version:**
3.3 RC 9

**OS/device including version:**
Android

**Issue description:**
I can't argue that this is a bug, but I got an unpleasant feeling from the message I received on Google Play.
Google says it will remove my games that use file access, and I'm not sure if I know what to do about it.

More details are written here: https://support.google.com/googleplay/android-developer/answer/10467955

Also, I did not find the necessary settings when exporting, perhaps some explanation is required on working with flags or the manifest directly.

### PR DESCRIPTION
**Pull request by m4gr3d**
Fixes #47910

### CODE REVIEW

File: platform/android/export/export.cpp [hunk 1]
```diff
@@ -768,6 +768,10 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 		return OK;
 	}
 
+	bool _has_storage_permission(const Vector<String> &permissions) {
```
Comment (akien-mga) [inline platform/android/export/export.cpp:771]:
```suggestion
	bool _has_storage_permission(const Vector<String> &p_permissions) {
```

File: platform/android/export/gradle_export_util.h [hunk 2]
```diff
@@ -275,13 +275,16 @@ String _get_activity_tag(const Ref<EditorExportPreset> &p_preset) {
 	return manifest_activity_text;
 }
 
-String _get_application_tag(const Ref<EditorExportPreset> &p_preset) {
+String _get_application_tag(const Ref<EditorExportPreset> &p_preset, bool has_storage_permission) {
```
Comment (akien-mga) [inline platform/android/export/gradle_export_util.h:278]:
```suggestion
String _get_application_tag(const Ref<EditorExportPreset> &p_preset, bool p_has_storage_permission) {
```

File: platform/android/export/export.cpp (hunk 1 above)
Comment (m4gr3d) [reply platform/android/export/export.cpp]: Done!

File: platform/android/export/gradle_export_util.h (hunk 2 above)
Comment (m4gr3d) [reply platform/android/export/gradle_export_util.h]: Done!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by anisc** [comment]
with 3.2.3,  Would adding android:requestLegacyExternalStorage="false" to the android manifest under android/build in the app change anything? 
My app only permission is internet and I have no idea how to fix it, by may 5th it will be pulled of the google play store if I don't find a solution.

**Comment by m4gr3d** [comment]
@anisc You can add `tools:remove="android:requestLegacyExternalStorage"` to the manifest under `android/build` to remove it.
Here's the [link to the full documentation](https://developer.android.com/studio/build/manifest-merge#attribute_markers).

**Comment by anisc** [comment]
@m4gr3d thank you so much for returning to me.
Ok I will try it and let you know because I did try : android:requestLegacyExternalStorage="false" under <application> but nothing changed so I will try "remove" hopefully it will work.

**Comment by besyuser** [comment]
Hello @anisc, I have the same problem. 
Did the tools:remove="android:requestLegacyExternalStorage" in the manifest work for you?
Do you happen to know if the playstore updates you about the compliance of your app after submitting your new version?

**Comment by akien-mga** [comment]
You can use Godot 3.3 and you won't need to add any workaround to the manifest: https://godotengine.org/article/godot-3-3-has-arrived

**Comment by anisc** [comment]
@besyuser As @akien-mga said i'm using godot 3.3 published the game but the warning is not gone on the console, so at this point I'm just waiting and crossing my fingers.
I did speak with customer service and they said it is normal for the warning to stay as a form of "archive"... 😕😕

**Comment by besyuser** [comment]
Thank you guys! Yesterday I published a version with the android manifest change. And today I published another one using Godot 3.3
We should be OK then.

**Pull request merged**